### PR TITLE
add AuthService

### PR DESCRIPTION
**Pull request by FirdavsAsadov**

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/Backend Project/Application/Identity/Service/IAuthService.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Backend_Project.Application.Identity.Models;
+
+namespace Backend_Project.Application.Identity.Service
+{
+    public interface IAuthService
+    {
+        ValueTask<bool> RegisterAsync(RegistrationDetails registrationDetails, string password);
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Application/Identity/Service/IAuthService.cs:7]: password parameter is redundant

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AccessTokenGeneratorService.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                expires: DateTime.Now.AddMinutes(_jwtSettings.ExpirationTimeInMinutes),
+                signingCredentials: credintials);
+        }
+
+        public List<Claim> GetClaims(User user)
+        {
+            return new List<Claim>()
+            {
+                new (ClaimTypes.Email, user.EmailAddress),
+                new (ClaimConstant.UserId, user.Id.ToString())
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AccessTokenGeneratorService.cs:50]: Add user role in the claims

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        public AuthService(IAccountService accountService, IAccessTokenGeneratorService accessTokenGeneratorService, IDataContext dataContext)
+        {
+            _accountService = accountService;
+            _accessTokenGeneratorService = accessTokenGeneratorService;
+            _dataContext = dataContext;
+        }
+
+        public async ValueTask<bool> RegisterAsync(RegistrationDetails registrationDetails, string password)
+        {
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs:23]: password parameter is redundant

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+using Backend_Project.Domain.Exceptions.EntityExceptions;
+using Backend_Project.Persistence.DataContexts;
+
+namespace Backend_Project.Infrastructure.Services.AccountServices
+{
+    public class AuthService : IAuthService
+    {
+        private readonly IAccountService _accountService;
+        private readonly IAccessTokenGeneratorService _accessTokenGeneratorService;
+        private readonly IDataContext _dataContext;
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs:13]: IDataContext AuthService ichiga inject qilinmidi

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            };
+
+            await _accountService.CreateUserAsync(newUser, password);
+
+            return true;
+        }
+
+        public ValueTask<string> LoginAsync(LoginDetails loginDetails)
+        {
+            var exsistingUser = _dataContext.Users.FirstOrDefault(user => user.EmailAddress == loginDetails.EmailAddress);
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs:39]: AccountService dan foydalanib userlarni query qilish kerak.

**Review by Ilxom-22 — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            return true;
+        }
+
+        public ValueTask<string> LoginAsync(LoginDetails loginDetails)
+        {
+            var exsistingUser = _accountService.GetUserByEmailAddress(loginDetails.EmailAddress);
+
+            if (exsistingUser == null)
+                throw new EntityNotFoundException("This User does not exsist!!");
+
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs:40]: EntityNotFoundException generic qilib yuborin

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AccountService.cs [hunk 7]
```diff
@@ -33,13 +34,24 @@ public async ValueTask<User> CreateUserAsync(User user, string password)
...
             UserId = createdUser.Id,
+
             Password = _passwordHasher.Hash(password)
         };
 
         await _userCredentialsService.CreateAsync(userCredentials);
 
         return createdUser;
     }
+    public User GetUserByEmailAddress(string emailAddress)
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AccountService.cs:45]: Sherdaham qator tashlab keting

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+            _accountService = accountService;
+            _accessTokenGeneratorService = accessTokenGeneratorService;
+        }
+
+        public async ValueTask<bool> RegisterAsync(RegistrationDetails registrationDetails)
+        {
+            var newUser = new User
+            {
+                Id = Guid.NewGuid(),
+                FirstName = registrationDetails.FirstName,
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AuthService.cs:24]: Id ni bazani o'zi yozib beradi

File: src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AccountService.cs [hunk 9]
```diff
@@ -33,13 +34,24 @@ public async ValueTask<User> CreateUserAsync(User user, string password)
         var userCredentials = new UserCredentials
         {
             UserId = createdUser.Id,
+
             Password = _passwordHasher.Hash(password)
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/Services/AccountServices/AccountService.cs:38]: 37 chi Qatorni o'chirib tashlan

**Review by Abdurahmon0412 — CHANGES_REQUESTED** [review]

**Review by Ilxom-22 — APPROVED** [review]

**Review by Abdurahmon0412 — APPROVED** [review]

**Pull request merged**